src/views/ActivitiesAll/index.js

```diff
@@ -58,10 +59,35 @@ export default class GordonActivitiesAll extends Component {
         activities,
         allActivities: activities,
         myInvolvements: myInvolvements,
-        loading: false,
+        loading: true,
         sessions,
         types,
       });
+
+      if (activities.length === 0) {
+        var recentSession;
+        recentSession = this.state.sessions[0].SessionCode;
+        this.setState({ session: recentSession, currentSession: sessionCode });
+        const [activities, types, sessions] = await Promise.all([
+          activity.getAll(recentSession),
+          activity.getTypes(recentSession),
+          session.getAll(),
+        ]);
+        const myInvolvements = await user.getCurrentMembershipsWithoutGuests(profile.ID);
+
+        this.setState({
+          profile,
+          activities,
+          allActivities: activities,
+          myInvolvements: myInvolvements,
+          loading: false,
+          sessions,
+          types,
+        });
+      }
+      this.setState({
+        loading: false,
```

Comment:
If we set loading:false in the setState directly above this one, why do we need to do it again here?